### PR TITLE
[FLINK-34323][table-planner] Fix param name in session window tvf when using named params

### PR DESCRIPTION
**Pull request by xuyangzhong**
## What is the purpose of the change

Currently, when user uses session window tvf by named params, an exception will be thrown.
This pr is try to align the param name in Flip-145 about session window tvf. 

## Brief change log

  - *align the param name from calcite `SIZE` to `GAP` in Flip-145 about session window tvf*
  - *Add tests to verify it*

## Verifying this change

Some tests are added to verify it.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented?

### CODE REVIEW

**Comment by xuyangzhong** [comment]
The failure is caused by https://issues.apache.org/jira/browse/FLINK-34272

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala [hunk 1]
```diff
@@ -257,4 +257,48 @@ class WindowTableFunctionTest extends TableTestBase {
...
+    val sql =
+      """
+        |SELECT *
+        |FROM TABLE(SESSION(TABLE MyTable PARTITION BY (b, a), DESCRIPTOR(rowtime), INTERVAL '15' MINUTE))
+        |""".stripMargin
+    util.verifyRelPlan(sql)
+  }
+
+  @Test
+  def testSessionTVFWithNamedParams(): Unit = {
```
Comment (hackergin) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala:291]: If the order of parameters is changed, it seems that verification cannot be passed now?
```sql
     SESSION(
         TIMECOL => DESCRIPTOR(rowtime),
         DATA => TABLE MyTable PARTITION BY (b, a),
         GAP => INTERVAL '15' MINUTE))
```

Do we also need to support this?

**Review by hackergin — COMMENTED** [review]
@xuyangzhong Thanks for the contribution, only one minor question.

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala (hunk 1 above)
Comment (xuyangzhong) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala:291]: I have double checked the Flip-145 and the user doc. I only find that `the DATA param must be the first`. That means we should allow the order changes about param DATA and GAP. But I think it's better to open a separate Jira for it. This pr focuses on the wrong PARAM name between `SIZE` in Calcite and `GAP` in Flink. WDYT?

Comment (hackergin) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala:291]: +1, we can handle it in a separate PR, it seems that other types of TVFs may also have similar issues.

Comment (xuyangzhong) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala:291]: I also add a related test for it.

**Review by hackergin — APPROVED** [review]
Thanks for the update, LGTM.

**Review by LadyForest — APPROVED** [review]
Thanks for the fix, LGTM

**Pull request merged**